tools/throttle/scripts/listCdnVersions.ps1

```diff
@@ -0,0 +1,114 @@
+    [string] $cdnStorePath = "cdnstoragename",          # Identifies the target Azure Storage account (by name)
+    [string] $subscriptionId = $null,                   # Identifies the target Azure Subscription Id (if not encoded in the cdnStorePath)
+    [string] $resourceGroup = $null,                    # Identifies the target Azure Subscription Resource Group (if not encoded in the cdnStorePath)
+    [string] $sasToken = $null,                         # The SAS Token to use rather than using or attempting to login
+    [string] $logPath = $null,                          # The location where logs should be written
+    [switch] $showFiles = $false,                       # Show the individual files with details as well
+    [switch] $activeOnly = $false,                      # Only show the active (deployed) versions
+    [switch] $testOnly = $false,                        # Uploads to a "tst" test container on the storage account
+    [switch] $cdn = $false                              # (No longer used -- kept for now for backward compatibility)
+)
+
+Import-Module -Force -Name "../../../common/publish/Logging"
+Import-Module -Force -Name "../../../common/publish/AzureStorageHelper"
+
+[hashtable]$global:connectDetails = @{}
+$global:connectDetails.storeContainer = $storeContainer
+$global:connectDetails.cdnStorePath = $cdnStorePath
+$global:connectDetails.resourceGroup = $resourceGroup
+$global:connectDetails.storeName = $null                              # The endpoint needs to the base name of the endpoint, not the full URL (eg. “my-cdn” rather than “my-cdn.azureedge.net”)
+$global:connectDetails.subscriptionId = $subscriptionId
+$global:connectDetails.sasToken = $sasToken
+$global:connectDetails.storageContext = $null
+$global:connectDetails.testOnly = $testOnly
+
+Function Write-LogParams 
+{
+    $logDir = Get-LogPath
+
+    Write-Log "Container         : $container"
+    Write-Log "Storage Container : $storeContainer"
+    Write-Log "Store Path        : $($global:connectDetails.cdnStorePath)"
+    Write-Log "Log Path          : $logDir"
+    Write-Log "Show Files        : $showFiles"
+    Write-Log "Test Mode         : $testOnly"
+    
+    if ([string]::IsNullOrWhiteSpace($global:connectDetails.sasToken) -eq $true) {
+        Write-Log "Mode      : User-Credentials"
+    } else {
+        Write-Log "Mode      : Sas-Token"
+    }
+}
+
+Function Validate-Params
+{
+    # Validate parameters
+    if ([string]::IsNullOrWhiteSpace($container) -ne $true -and "beta","next","public", "dev", "nightly" -NotContains $container) {
+        Write-LogFailure "[$($container)] is not a valid value, must be beta, next or public"
+    }
+}
+
+Function Get-AllVersionFiles(
+    [system.collections.generic.dictionary[string, system.collections.generic.list[hashtable]]] $files,
+    [string] $storagePath
+) {
+    Get-VersionFiles $files "$storagePath/ext" "ai.throttle." $null
```

Comment:
We will actually put this in the $storagePath -- so remove the /ext (which is for extensions)